File: create_tables/land_size.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table land_size';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute("""
    create table land_size(
        LAND_RIGHT_SUBTYPE VARCHAR(30) NOT NULL,
        LAND_RIGHT_ID VARCHAR(40) PRIMARY KEY NOT NULL,
        SIZE_TYPE VARCHAR(40) NOT NULL,
        SIZE_SEQ_NO INT,
        ACTIVE_IND VARCHAR(1),
        BUSINESS_ASSOCIATE_ID VARCHAR(40),
        EFFECTIVE_DATE TIMESTAMP(0),
        EXPIRY_DATE TIMESTAMP(0),
        GROSS_SIZE DECIMAL(20,10),
        INTEREST_SET_ID VARCHAR(40),
        INTEREST_SET_SEQ_NO INT,
        NET_SIZE DECIMAL(20,10),
        PARTNER_OBS_NO INT,
        PPDM_GUID VARCHAR(38),
        REMARK VARCHAR(2000),
        SIZE_OUOM VARCHAR(40),
        SOURCE VARCHAR(40),
        ROW_CHANGED_BY VARCHAR(30),
        ROW_CHANGED_DATE TIMESTAMP(0),
        ROW_CREATED_BY VARCHAR(30),
        ROW_CREATED_DATE TIMESTAMP(0),
        ROW_EFFECTIVE_DATE TIMESTAMP(0),
        ROW_EXPIRY_DATE TIMESTAMP(0),
        ROW_QUALITY VARCHAR(40))""")

File: postgres_39/all_struct_with_default/LandSize.go
package dto

type Land_size struct {
	Land_right_subtype    string   `json:"land_right_subtype" default:"source"`
	Land_right_id         string   `json:"land_right_id" default:"source"`
	Size_type             string   `json:"size_type" default:"source"`
	Size_seq_no           int      `json:"size_seq_no" default:"1"`
	Active_ind            *string  `json:"active_ind" default:""`
	Business_associate_id *string  `json:"business_associate_id" default:""`
	Effective_date        *string  `json:"effective_date" default:""`
	Expiry_date           *string  `json:"expiry_date" default:""`
	Gross_size            *float64 `json:"gross_size" default:""`
	Interest_set_id       *string  `json:"interest_set_id" default:""`
	Interest_set_seq_no   *int     `json:"interest_set_seq_no" default:""`
	Net_size              *float64 `json:"net_size" default:""`
	Partner_obs_no        *int     `json:"partner_obs_no" default:""`
	Ppdm_guid             *string  `json:"ppdm_guid" default:""`
	Remark                *string  `json:"remark" default:""`
	Size_ouom             *string  `json:"size_ouom" default:""`
	Source                *string  `json:"source" default:""`
	Row_changed_by        *string  `json:"row_changed_by" default:""`
	Row_changed_date      *string  `json:"row_changed_date" default:""`
	Row_created_by        *string  `json:"row_created_by" default:""`
	Row_created_date      *string  `json:"row_created_date" default:""`
	Row_effective_date    *string  `json:"row_effective_date" default:""`
	Row_expiry_date       *string  `json:"row_expiry_date" default:""`
	Row_quality           *string  `json:"row_quality" default:""`
}
